rholang/src/rust/interpreter/compiler/normalizer/processes/p_send_sync_normalizer.rs

```diff
@@ -0,0 +1,224 @@
+use std::collections::HashMap;
+use models::rhoapi::Par;
+use super::exports::*;
+use crate::rust::interpreter::compiler::normalize::{normalize_match_proc, ProcVisitInputs, ProcVisitOutputs};
+use crate::rust::interpreter::compiler::rholang_ast;
+use crate::rust::interpreter::compiler::rholang_ast::{Name, Proc, SyncSendCont};
+use crate::rust::interpreter::compiler::rholang_ast::{NameDecl, ProcList};
+use crate::rust::interpreter::errors::InterpreterError;
+use uuid::Uuid;
+
+pub fn normalize_p_send_sync(
+  name: &Name,
+  messages: &ProcList,
+  cont: &SyncSendCont,
+  line_num: usize,
+  col_num: usize,
+  input: ProcVisitInputs,
+  env: &HashMap<String, Par>,
+) -> Result<ProcVisitOutputs, InterpreterError> {
+
+  let identifier = Uuid::new_v4().to_string();
+  let name_var: rholang_ast::Proc = Proc::Eval(rholang_ast::Eval {
+    name: rholang_ast::Name::ProcVar(Box::new(rholang_ast::Proc::Var(rholang_ast::Var {
+      name: identifier.clone(),
+      line_num,
+      col_num,
+    })),),
+    line_num,
+    col_num,
+  });
+
+
+  let send: Proc = {
+    let mut listproc =
+      messages.procs.clone();
+
+    listproc.insert(0, name_var.clone());
+
+    Proc::Send {
+      name: name.clone(),
+      send_type: rholang_ast::SendType::Single {
+        line_num,
+        col_num,
+      },
+      inputs: ProcList {
+        procs: listproc,
+        line_num,
+        col_num,
+      },
+      line_num: messages.line_num,
+      col_num: messages.col_num,
+    }
+  };
+
+  let receive: Proc = {
+    let list_name = rholang_ast::Names {
+      names: vec![rholang_ast::Name::ProcVar(Box::new(rholang_ast::Proc::Wildcard {
+        line_num,
+        col_num,
+      }))],
+      cont: None,
+      line_num,
+      col_num,
+    };
+
+    let linear_bind_impl: rholang_ast::LinearBind = rholang_ast::LinearBind {
+      names: list_name,
+      input: rholang_ast::Source::Simple {
+        name: rholang_ast::Name::ProcVar(Box::new(name_var)), // This I think needs to be Proc::Var( Var { .. }), not Proc::Eval inside Name::ProcVar
```

Comment:
it's `Proc::Eval` at line 22 because of it was [PEval in Scala](https://github.com/F1R3FLY-io/f1r3fly/blob/andrii/rholang_rust/rholang/src/main/scala/coop/rchain/rholang/interpreter/compiler/normalizer/processes/PSendSynchNormalizer.scala#L20)

is that correct?